src/ArchimedeanCopula.jl

```diff
@@ -198,3 +187,101 @@ function Distributions._rand!(rng::Distributions.AbstractRNG, ::ArchimedeanCopul
     return A
 end
 
+function Distributions._rand!(rng::Distributions.AbstractRNG, C::ClaytonCopula, x::AbstractVector{<:Real})
+    x[:] = inverse_rosenblatt(C, rand(rng, length(x)))
+    return x
+end
+function Distributions._rand!(rng::Distributions.AbstractRNG, C::ClaytonCopula, A::DenseMatrix{<:Real})
+    A[:] = inverse_rosenblatt(C, rand(rng, size(A)...))
+    return A
+end
+
+function rosenblatt(C::ArchimedeanCopula{d,TG}, u::AbstractMatrix{<:Real}) where {d,TG}
+    @assert d == size(u, 1)
+
+    U = zeros(eltype(u), size(u))
+    U[1, :] = u[1, :]
+
+    for j in 2:d
+        U[j, :] .=
+            ϕ⁽ᵏ⁾.(C.G, Val(j - 1), reduce(+, [ϕ⁻¹.(C.G, u[k, :]) for k in 1:j])) ./ ϕ⁽ᵏ⁾.(C.G, Val(j - 1), reduce(+, [ϕ⁻¹.(C.G, u[k, :]) for k in 1:(j - 1)]))
+    end
+
+    return U
+end
+
+function rosenblatt(
+    C::ArchimedeanCopula{d,TG}, u::AbstractMatrix{<:Real}
+) where {d,TG<:ClaytonGenerator}
+    @assert d == size(u, 1)
+
+    U = zeros(eltype(u), size(u))
+    U[1, :] = u[1, :]
+
+    for j in 2:d
+        U[j, :] .=
+            (
+                (1 .- j .+ sum(u[1:j, :] .^ (-C.G.θ); dims=1)[:]) ./
+                (2 .- j .+ sum(u[1:(j - 1), :] .^ (-C.G.θ); dims=1)[:])

```

Comment:
Here you are recomputing `u^θ` a lot of times. maybe pre-compute it ?

Reply:
Even without recomputing u^\theta, still some of these sum looks like too much work